inspirehep/modules/cache/utils.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2017-2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from contextlib import contextmanager
+
+from redis import StrictRedis
+from redis_lock import Lock
+
+from inspire_utils.config import load_config
+
+from .errors import RedisLockError
+
+
+@contextmanager
+def redis_locking_context(lock_name, expire=120, auto_renewal=True):
+    """Locked Context Manager to perform operations on Redis."""
+    if not lock_name:
+        raise RedisLockError('Lock name not specified.')
+
+    config = load_config()
```

Comment:
😕 isn't part of ``inspirehep`` app? why we load config differently and don't use the ``current_app``?